src/controllers/authControllers.ts

```diff
@@ -1,20 +1,37 @@
-const { User } = require("../models/User");
-const bcrypt = require("bcrypt");
+import { User } from "../models/User";
+
+import bcrypt from "bcrypt";
+import Joi from "joi";
+import passwordComplexity from "joi-password-complexity";
+
 const label = "Password";
-const Joi = require("joi");
-const passwordComplexity = require("joi-password-complexity");
 
-const createError = (msg, code = 403) => {
-  const err = new Error(msg);
-  //console.log(message);
-  err.code = code;
-  //console.log(err);
+// const createError = (msg, code = 403) => {
+//   const err = new Error(msg);
+//   //console.log(message);
+//   err.code = code;
+//   //console.log(err);
+//   return err;
+// };
+class customError extends Error {
```

Comment:
Abstract the custom error to a single module that can be used by all controllers.